pkg/options/options.go

```diff
@@ -12,64 +12,66 @@ import (
-	OverrideNextHop         bool
-	PeerASNs                []uint
-	PeerMultihopTtl         uint8
-	PeerPasswords           []string
-	PeerPorts               []uint
-	PeerRouters             []net.IP
-	RouterId                string
-	RoutesSyncPeriod        time.Duration
-	RunFirewall             bool
-	RunRouter               bool
-	RunServiceProxy         bool
-	Version                 bool
-	VLevel                  string
+	AdvertiseClusterIp             bool
+	AdvertiseExternalIp            bool
+	AdvertiseNodePodCidr           bool
+	AdvertiseLoadBalancerIp        bool
+	BGPGracefulRestart             bool
+	BGPGracefulRestartDeferralTime time.Duration
+	BGPPort                        uint16
+	CacheSyncTimeout               time.Duration
+	CleanupConfig                  bool
+	ClusterAsn                     uint
+	ClusterCIDR                    string
+	DisableSrcDstCheck             bool
+	EnableCNI                      bool
+	EnableiBGP                     bool
+	EnableOverlay                  bool
+	EnablePodEgress                bool
+	EnablePprof                    bool
+	FullMeshMode                   bool
+	OverlayType                    string
+	GlobalHairpinMode              bool
+	HealthPort                     uint16
+	HelpRequested                  bool
+	HostnameOverride               string
+	IPTablesSyncPeriod             time.Duration
+	IpvsSyncPeriod                 time.Duration
+	IpvsGracefulPeriod             time.Duration
+	IpvsGracefulTermination        bool
+	Kubeconfig                     string
+	MasqueradeAll                  bool
+	Master                         string
+	MetricsEnabled                 bool
+	MetricsPath                    string
+	MetricsPort                    uint16
+	NodePortBindOnAllIp            bool
+	OverrideNextHop                bool
+	PeerASNs                       []uint
+	PeerMultihopTtl                uint8
+	PeerPasswords                  []string
+	PeerPorts                      []uint
+	PeerRouters                    []net.IP
+	RouterId                       string
+	RoutesSyncPeriod               time.Duration
+	RunFirewall                    bool
+	RunRouter                      bool
+	RunServiceProxy                bool
+	Version                        bool
+	VLevel                         string
 	// FullMeshPassword    string
 }
 
 func NewKubeRouterConfig() *KubeRouterConfig {
 	return &KubeRouterConfig{
-		CacheSyncTimeout:   1 * time.Minute,
-		IpvsSyncPeriod:     5 * time.Minute,
-		IPTablesSyncPeriod: 5 * time.Minute,
-		IpvsGracefulPeriod: 30 * time.Second,
-		RoutesSyncPeriod:   5 * time.Minute,
-		EnableOverlay:      true,
-		OverlayType:        "subnet",
+		CacheSyncTimeout:               1 * time.Minute,
+		IpvsSyncPeriod:                 5 * time.Minute,
+		IPTablesSyncPeriod:             5 * time.Minute,
+		IpvsGracefulPeriod:             30 * time.Second,
+		RoutesSyncPeriod:               5 * time.Minute,
+		BGPGracefulRestartDeferralTime: 360 * time.Second,
```

Comment:
How was this default decided?

Reply:
FWIW, it's the gobgp default, so is fully backward compatible with prior kube-router releases.

Reply:
ah, good to know thanks!